skyportal/tests/frontend/test_upload_observations.py

```diff
@@ -0,0 +1,70 @@
+import os
+import uuid
+import time
+import pandas as pd
+from regions import Regions
+
+from skyportal.tests import api
+
+
+def test_upload_observations(driver, super_admin_user, super_admin_token):
+
+    telescope_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'telescope',
+        data={
+            'name': telescope_name,
+            'nickname': telescope_name,
+            'lat': 0.0,
+            'lon': 0.0,
+            'elevation': 0.0,
+            'diameter': 10.0,
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+    telescope_id = data['data']['id']
+
+    fielddatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Fields.csv'
+    regionsdatafile = f'{os.path.dirname(__file__)}/../../../data/ZTF_Region.reg'
+
+    instrument_name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'instrument',
+        data={
+            'name': instrument_name,
+            'type': 'imager',
+            'band': 'Optical',
+            'filters': ['ztfr'],
+            'telescope_id': telescope_id,
+            'field_data': pd.read_csv(fielddatafile)[:5].to_dict(orient='list'),
+            'field_region': Regions.read(regionsdatafile).serialize(format='ds9'),
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # wait for the fields to populate
+    time.sleep(15)
+
+    driver.get(f"/become_user/{super_admin_user.id}")
+    driver.get("/observations/")
+
+    filename = "sample_observation_data_upload.csv"
+
+    attachment_file = driver.wait_for_xpath('//input[@type="file"]')
+    attachment_file.send_keys(
+        os.path.join(
+            os.path.dirname(os.path.dirname(__file__)),
+            'data',
+            filename,
+        ),
+    )
+
+    driver.wait_for_xpath(f'//*[contains(., "{filename}")]')
+    submit_button_xpath = '//button[contains(.,"Submit")]'
+    driver.click_xpath(submit_button_xpath, scroll_parent=True)
```

Comment:
Can you now test that those observations are indeed there/uploaded?

Reply:
Please also test behavior when you upload 
 1) malformed observations
 2) observations without airmass and seeing
 3) identical observations to what was loaded by someone else.

(I dont know what behavior you expect in 3 above, should be documented anyway).

Reply:
Added. The identical observations one I do not currently test for as that is a log message, as opposed to something returned to the frontend. (just with the baselayer log). I made a few changes to the doc to make it clear what is actually required.